### PR TITLE
Feature/edit activity refactor

### PR DESCRIPTION
**Pull request by kejmil01**
TODO:
- tests

### CODE REVIEW

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 1]
```diff
@@ -95,6 +83,27 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
+        gpsEstablisher = new GpsEstablisher(locationManager, googleLocationApiEstablisher, this);
+        gpsEstablisher.setListener(new GpsEstablisher.Listener() {
+            @Override
+            public void onGpsEstablished() {
+                openPlacePicker();
+            }
+
+            @Override
+            public void onGoogleLocationApiConnectionFailure() {
+                Toast.makeText(EditActivity.this, R.string.google_api_connection_error, Toast.LENGTH_SHORT).show();
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java]: Make it snackbar please

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+class GpsEstablisher {
+    private static final String TAG = GpsEstablisher.class.getSimpleName();
+
+    private static final int GPS_REQUEST_CODE_DEFAULT = 1234;
+    private static final long LOCATION_REQUEST_INTERVAL = 100;
+    private static final long LOCATION_REQUEST_FASTEST_INTERVAL = 100;
+
+    private GoogleLocationApiEstablisher googleLocationApiEstablisher;
+    private LocationManager locationManager;
+    private Activity activity;
```
Comment (michal1990) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: do you really need activity in this place? If yes, better will be weak reference notation

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import java.lang.ref.WeakReference;
+
+import javax.annotation.Nullable;
+
+class GoogleLocationApiEstablisher {
+
+    private static final String TAG = GoogleLocationApiEstablisher.class.getSimpleName();
+
+    private Listener listener;
+    private GoogleApiClient.Builder googleApiClientBuilder;
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java]: final

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    @Nullable
+    private GoogleApiClient googleApiClient;
+    private WeakReference<Context> contextWeakReference;
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java]: final

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+import javax.annotation.Nullable;
+
+class GpsEstablisher {
+    private static final String TAG = GpsEstablisher.class.getSimpleName();
+
+    private static final int GPS_REQUEST_CODE_DEFAULT = 1234;
+    private static final long LOCATION_REQUEST_INTERVAL = 100;
+    private static final long LOCATION_REQUEST_FASTEST_INTERVAL = 100;
+
+    private GoogleLocationApiEstablisher googleLocationApiEstablisher;
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: final

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    private LocationManager locationManager;
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: final

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java (hunk 2 above)
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: final

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        locationRequest.setFastestInterval(LOCATION_REQUEST_FASTEST_INTERVAL);
+        locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+        return locationRequest;
+    }
+
+    public void setListener(@Nullable Listener listener) {
+        this.listener = listener;
+    }
+
+    interface Listener {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: Maybe something more? Listener is not enough IMO

**Review by BartSoftware — CHANGES_REQUESTED** [review]

**Review by michal1990 — APPROVED** [review]

**Review by BartSoftware — APPROVED** [review]

**Pull request merged**